v2/event/event_marshal.go

```diff
@@ -76,7 +76,7 @@ func WriteJson(in *Event, writer io.Writer) error {
 		if eventContext.Time != nil {
 			stream.WriteMore()
 			stream.WriteObjectField("time")
-			stream.WriteString(eventContext.Time.String())
+			stream.WriteString(eventContext.Time.Format(time.RFC3339))
```

Comment:
I ran an example w/ Go 1.22 (same w/ Go 1.21) here: https://go.dev/play/p/NcMTqPu75fe

And I get:

```
2009-11-10 23:00:00 +0000 UTC m=+0.000000001
2009-11-10T23:00:00Z
```

Which could be considered a breaking change. WDYT?

cc/ @duglin

Reply:
I'm ok with the breaking change. I'm more bothered that:
1 - it wasn't spec compliant before and no one noticed
2 - not a single testcase needed to be changed

I agree with splitting the upgrade stuff into a separate PR. But can we also get at least one testcase added that checks the format of the `time` attribute and that it adheres to the CE spec (RFC3339) ?

Reply:
I'll get to work on it! Thank you so much for the feedback!

Reply:
Cleaned up the PR. Currently working on adding test cases:)